src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -2716,10 +2934,7 @@ - (void)_setupSubscriptionWithReason:(NSString *)reason
 
                            // Callback and ClusterStateCache and ReadClient will be deleted
                            // when OnDone is called.
-                           os_unfair_lock_lock(&self->_lock);
-                           self->_currentReadClient = readClient.get();
-                           self->_currentSubscriptionCallback = callback.get();
-                           os_unfair_lock_unlock(&self->_lock);
+                           [matterCPPObjectsHolder setReadClient:readClient.get() subscriptionCallback:callback.get()];
```

Comment:
Hmm.  So we used to lock here, but maybe in practice that was not needed because all the accesses involved are on the Matter queue?

Reply:
See https://github.com/project-chip/connectedhomeip/issues/37420